polaris-react/src/components/IndexTable/IndexTable.stories.tsx

```diff
@@ -4431,3 +4617,1477 @@ export function WithSubHeadersNonSelectable() {
+      selectableRows.findIndex(
+        (row) => row.id === products[products.length - 1].id,
+      ),
+    ];
+
+    const disabled = products.every(({disabled}) => disabled);
+
+    return (
+      <Fragment key={subheaderId}>
+        <IndexTable.Row
+          rowType="subheader"
+          selectionRange={rowRange}
+          id={`Subheader-${index}`}
+          position={position}
+          selected={selected}
+          disabled={disabled}
+          accessibilityLabel={`Select all products wich has color ${color}`}
+        >
+          <IndexTable.Cell>
+            <Thumbnail
+              source="https://burst.shopifycdn.com/photos/black-leather-choker-necklace_373x@2x.jpg"
+              size="medium"
+              alt="Black choker necklace"
+            />
+          </IndexTable.Cell>
+          <IndexTable.Cell id={subheaderId}>
+            <Text as="span" fontWeight="semibold">
+              {color}
+            </Text>
+          </IndexTable.Cell>
+          <IndexTable.Cell />
+          <IndexTable.Cell />
+        </IndexTable.Row>
+        {products.map(
+          ({id, size, quantity, price, position, disabled}, rowIndex) => {
+            return (
+              <IndexTable.Row
+                rowType="child"
+                key={rowIndex}
+                id={id}
+                position={position}
+                selected={selectedResources.includes(id)}
+                disabled={disabled}
+              >
+                <IndexTable.Cell>
+                  <Thumbnail
+                    source="https://burst.shopifycdn.com/photos/black-leather-choker-necklace_373x@2x.jpg"
+                    size="small"
+                    alt="Black choker necklace"
+                  />
+                </IndexTable.Cell>
+                <IndexTable.Cell>
+                  <Text variant="bodyMd" as="span">
+                    {size}
+                  </Text>
+                </IndexTable.Cell>
+                <IndexTable.Cell>
+                  <Text as="span" numeric>
+                    {price}
+                  </Text>
+                </IndexTable.Cell>
+                <IndexTable.Cell>
+                  <Text as="span" alignment="end" numeric>
+                    {quantity}
+                  </Text>
+                </IndexTable.Cell>
+              </IndexTable.Row>
+            );
+          },
+        )}
+      </Fragment>
+    );
+  });
+
+  return (
+    <LegacyCard>
+      <IndexTable
+        onSelectionChange={handleSelectionChange}
+        selectedItemsCount={
+          allResourcesSelected ? 'All' : selectedResources.length
+        }
+        resourceName={resourceName}
+        itemCount={rows.length}
+        headings={columnHeadings as IndexTableProps['headings']}
+        selectable={false}
+      >
+        {rowMarkup}
+      </IndexTable>
+    </LegacyCard>
+  );
+}
+
+export function WithNestedRowsWithThumbnailsOneCellSelectable() {
```

Comment:
Just noticed that there is too much distance between image and variant title after  snapshot was applied for variants card, going to fix it.
<img width="537" alt="image" src="https://github.com/Shopify/polaris/assets/104942025/db0cf393-82d2-4220-b15e-5f98d05bab13">